src/mondefense.cpp

```diff
@@ -150,8 +150,12 @@ void mdefense::return_fire( monster &m, Creature *source, const dealt_projectile
     if( m.is_dead_state() ) {
         return;
     }
+    // No return fire for quiet or completely silent projectiles (bows, throwing etc).
+    if( dynamic_cast<player *>( source )->weapon.gun_noise().volume < 6 ) {
```

Comment:
This will invoke UB if the source is not actually a player object (e.g. another monster). The dynamic cast will yield a nullpointer, which gets dereferenced.